generate_figures.py

```diff
@@ -105,19 +107,22 @@ def comma_seperators(x, pos):
 
 
 if __name__ == '__main__':
-    path = sys.argv[1] if len(sys.argv) > 1 else 'benchmark.csv'
-    system_type = sys.argv[2] if len(sys.argv) > 2 else 'power'
-    flags = sys.argv[3:]
-    show_variance = 'no-variance' not in flags
+    parser.add_argument('path', help='directory with files to process or single file')
+    parser.add_argument('--system', help='system to plot results for', choices=['intel', 'power'], default='intel')
```

Comment:
There should not be a default value.
This value should be required!
If you really want a default then use POWER ;)